### ISSUE TITLE
AG - Upgrade Camel K to latest version on all managed Openshift clusters

### ISSUE THREAD

**Issue by wmhutchison** [BCDevOps/developer-experience#5112]
**Describe the issue**
AG had sent an email to the Platform Services Team mailbox who then passed it along to Platform Operations. Recent ACS security scans revealed some vulnerabilities, which when coupled with AG opening a case with Red Hat on the matter. Red Hat's suggestion was that the Camel K Operator needs to be upgraded.

While Platform Operations did post on the case asking if the specific new version of this operator will in fact address the security concerns, the team will proceed with upgrading in all LAB clusters since AG has already expressed interest to help test this. 

**Additional context**
Since we are currently running v1.10.6 for the Camel K operator and the latest available is v1.10.7, this should be a relatively simple upgrade with no dependen y issues involved. Adjusting CCM manifest so that LAB and PROD versioning are separated out for this event since right now all clusters run the same version.

**How does this benefit the users of our platform?**
Ensuring all software components used in the formation of apps on the Openshift platform are kept up to date for security vulnerabilities ensures safe and secure services all around.

**Definition of done**
- [ ] Create a new CCM PR to promote newer Camel K Operator. Unit test to ensure manifests are generated as expected.
- [ ] Review/approve/merge PR. Apply specific operator in KLAB/CLAB/KLAB2 once CCM refreshes from the new commit.
- [ ] Ask AG to start testing in their LAB environments.
- [ ] Confirm if AG is satisfied with the results and start coordinating when they want the changes implemented (stick with CCM release and dates, create new RFC for different release schedule).
- [ ] Create new RFCs if required, execute after approval is received.
- [ ] Final confirmation with AG regarding this work addressing their concerns.

**Comment by wmhutchison** [comment]
https://github.com/bcgov-c/platform-gitops-gen/pull/924 created to promote the latest available version of Camel K in the LAB clusters. Approved/merged, now proceeding with updating KLAB/CLAB/KLAB2 before asking AG to start testing.

**Comment by wmhutchison** [comment]
New CCM PR created and unit-tested to ensure manifests only change where desired in LAB and not PROD. Have successfully upgraded in KLAB and CLAB. Some issues with OLM to look at in KLAB2. 

Will ask AG to test for now in available environments in KLAB/CLAB for the time being so they can start working on things.

**Comment by wmhutchison** [comment]
Found as well the case opened by AG with Red Hat and posted there with specific operator versioning info since their instructions to AG were generic in nature. We'd like Red Hat to also confirm that the new operator has what AG needs. This need to know is more for AG than acting as a blocker for Platform Operations, but if the newest operator does not address AG's needs, then this issue will need to remain with Red Hat until they have available a suitable version.

**Comment by wmhutchison** [comment]
Putting into Blocked for now, waiting on AG to finish their tests, and Red Hat support to request more info if required from the cluster side.

**Comment by wmhutchison** [comment]
Red Hat support did request more info, but was with respect to the built image used by AG, so AG is following up with Red Hat on that front.

**Comment by wmhutchison** [comment]
Nothing new to report. Red Hat appears to be treating this as a false report from ACS, and have asked AG to provide specifics from ACS regarding that. Platform Operations team will help if required, but with the ACS stakeholders now involved, they will likely fulfill that information request.

We have not received a request from AG to update PROD clusters to the same version as in LAB.

**Comment by StevenBarre** [comment]
AG needs help scanning their image via `roxctl`, passing the ticket to Nick.

**Comment by NickCorcoran** [comment]
Bruce worked with the AG team, but had a lot of issues getting scan to execute using roxctl.  Tickets were raised with Red Hat and business has also been in discussion w/ Red Hat support.  As we know that version has been updated, I think it's worth moving this to closed and if needed we can follow up in another ticket.

**Comment by StevenBarre** [comment]
Upgraded ACS to 4.5.4 and set up the newer v4 Scanners. RH TAM advised this is a known false positive and that the new scanner might fix that.

**Comment by StevenBarre** [comment]
No reply from user, followed up

**Comment by StevenBarre** [comment]
No reply from user, followed up

**Comment by NickCorcoran** [comment]
I have reached out to the AG MISO to follow up with team.

**Comment by StevenBarre** [comment]
Closed due to lack of feedback
